jenkins/job_manager/script/SettingsParser.groovy

```diff
@@ -0,0 +1,53 @@
+import org.yaml.snakeyaml.Yaml
+import groovy.mock.interceptor.MockFor
+
+class SettingsParser {
+    String yaml
+    Map config
+    List<Settings> settingsList = []
+
+    static Logger loggerMock
+
+    SettingsParser(String yaml, String branchName = null) {
+        initializeMocks()
+        parseYaml(yaml)
+
+        def trunk = config.trunks.find {it.name == "kinetic-devel"}
+        def parentName = null
+
+        if (config.branch) {
+            parentName = config.branch.parent
+        }
+
+        // if there are multiple settings, create list of maps
+        if (config.trunks && 'kinetic-devel' in config.trunks.name && trunk.settings.getClass() == ArrayList &&
```

Comment:
why kinetic-devel is hardcoded? Configuration should be indifferent what trunk name is